File: Data Structures & Algorithms/products-of-array-discluding-self/submission-0.cpp
class Solution {
public:
    vector<int> productExceptSelf(vector<int>& nums) {
        vector<int> result(nums.size(), 0);
        
        for(int i=0; i<nums.size(); i++){
            int left = 1;
            int right =1;
            for(int j=0; j<i; j++){
                left *= nums[j];
            }
            for(int k=nums.size()-1; k>i; k--){
                right *= nums[k];
            }
            result[i] = left * right;
        }
        return result;
    }
};

File: Data Structures & Algorithms/eating-bananas/submission-0.py
class Solution:
    def minEatingSpeed(self, piles: List[int], h: int) -> int:
        # min and max speed
        left, right = 1, max(piles)
    
        while left < right:
            mid = (left + right) // 2
            hour = sum(math.ceil(p / mid) for p in piles)
            if hour > h:
                # Too slow: increase the speed
                left = mid + 1
            else:
                right = mid
            
        return right
